eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/ConfigResolver.php

```diff
@@ -237,8 +274,39 @@ public function getDefaultScope()
         return $this->defaultScope ?: $this->siteAccess->name;
     }
 
+    /**
+     * @param string $scope The default "scope" aka siteaccess name, as opposed to the self::SCOPE_DEFAULT.
+     */
     public function setDefaultScope($scope)
     {
         $this->defaultScope = $scope;
+
+        // On scope change check if siteaccess has been updated so we can log warnings if there are any
+        if ($this->siteAccess->matchingType !== 'uninitialized') {
+            $this->logTooEarlyLoadedParams();
+        }
+    }
+
+    private function logTooEarlyLoadedParams()
+    {
+        if (empty($this->tooEarlyLoadedList)) {
+            return;
+        }
+
+        $logger = $this->container->get('logger');
```

Comment:
Please inject it :upside_down_face:

Reply:
No can do AFAIK, logger is not set yet at the stage where this is booted on console use  _(see ramblings on jira issue)_

Reply:
Any suggestion how this should be handled then? `logger` is a private service in Symfony 4 and it doesn't work.

Reply:
Not really.

You can strict mode instead perhaps, inject, and if logger is null thow.
That was not an option for existing releases.

Otherwise might be some way to inject a lazy service that depends on logger.